libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -207,45 +209,51 @@ public FileData withPlatform(Platform platform) {
         }
     }
 
-    private record PathSettingFileData(String setting, Mode mode, Platform platform) implements FileData {
+    private record PathSettingFileData(String setting, Mode mode, boolean ignoreUrl, Platform platform) implements FileData {
         @Override
         public Stream<Path> resolvePaths(PathLookup pathLookup) {
-            return resolvePathSettings(pathLookup, setting);
+            return resolvePathSettings(pathLookup, setting, ignoreUrl);
         }
 
         @Override
         public FileData withPlatform(Platform platform) {
             if (platform == platform()) {
                 return this;
             }
-            return new PathSettingFileData(setting, mode, platform);
+            return new PathSettingFileData(setting, mode, ignoreUrl, platform);
         }
     }
 
-    private record RelativePathSettingFileData(String setting, BaseDir baseDir, Mode mode, Platform platform)
+    private record RelativePathSettingFileData(String setting, BaseDir baseDir, Mode mode, boolean ignoreUrl, Platform platform)
         implements
             FileData,
             RelativeFileData {
         @Override
         public Stream<Path> resolveRelativePaths(PathLookup pathLookup) {
-            return resolvePathSettings(pathLookup, setting);
+            return resolvePathSettings(pathLookup, setting, ignoreUrl);
         }
 
         @Override
         public FileData withPlatform(Platform platform) {
             if (platform == platform()) {
                 return this;
             }
-            return new RelativePathSettingFileData(setting, baseDir, mode, platform);
+            return new RelativePathSettingFileData(setting, baseDir, mode, ignoreUrl, platform);
         }
     }
 
-    private static Stream<Path> resolvePathSettings(PathLookup pathLookup, String setting) {
+    private static Stream<Path> resolvePathSettings(PathLookup pathLookup, String setting, boolean ignoreUrl) {
+        Stream<String> result;
         if (setting.contains("*")) {
-            return pathLookup.settingGlobResolver().apply(setting).map(Path::of);
+            result = pathLookup.settingGlobResolver().apply(setting);
+        } else {
+            String path = pathLookup.settingResolver().apply(setting);
+            result = path == null ? Stream.of() : Stream.of(path);
+        }
+        if (ignoreUrl) {
+            result = result.filter(s -> s.toLowerCase(Locale.ROOT).startsWith("https://") == false);
         }
-        String path = pathLookup.settingResolver().apply(setting);
-        return path == null ? Stream.of() : Stream.of(Path.of(path));
+        return result.map(Path::of);
```

Comment:
In what scenario would it make sense to _not_ use ignore_url in a file entitlement? Wouldn't calling `Path.of()` explode on such a value if it weren't filtered out? Should this behavior be the default?

Reply:
Yes it would explode. But we don't support urls in all settings? These are very specific settings that allow a url to be present. I don't want to add leniency generally to all file settings.